practice-app/server/src/services/auth.js

```diff
@@ -0,0 +1,79 @@
+//We will do the authorization checks here !!
+const jwks = require("../services/jwkeys");
+const jwt = require('jsonwebtoken');
+const UserModel = require("../models/users/users.model");
+const axios = require("axios");
+require("dotenv").config();
+
+const AUTH0_DOMAIN = process.env.auth0_domain;
+const client_id = process.env.auth0_client_id;
+const client_secret = process.env.auth0_clientSecret;
+const audience = process.env.auth0_audience;
+const auth0_connection = process.env.auth0_connection
+
+const aud = process.env.auth0_audience
+
+const authorization = async (req, res, next) => {
+
+    try{
+        // JWT Validation
+        // Get pem certificates and verify the header token
+        const pem = jwks.getPem();
+        jwt.verify(req.headers.authorization, pem, (err, decoded) => {
+            tokenError = err;
+            decrytedData = decoded;
+        });
```

Comment:
What if the authorization header is not provided? Additionally, when I add a Bearer Authorization token, I get an error. I suspect that since req.headers.authorization is of form Bearer {token}, jwt cannot parse it. You need to extract token and then pass it to jwt.